service/storage_service.cc

```diff
@@ -598,6 +600,17 @@ future<> storage_service::sync_raft_topology_nodes(mutable_token_metadata_ptr tm
     }
 
     co_await when_all_succeed(sys_ks_futures.begin(), sys_ks_futures.end()).discard_result();
+
+    co_return nodes_to_notify;
+}
+
+future<> storage_service::notify_nodes_after_sync(nodes_to_notify_after_sync&& nodes_to_notify) {
+    for (auto [ip, host_id] : nodes_to_notify.left) {
+        co_await notify_left(ip, host_id);
+    }
+    for (auto ip : nodes_to_notify.joined) {
+        co_await notify_joined(ip);
```

Comment:
Orthogonal to this change, I wondered why notify_joined doesn't take a host_id as well.
It turns out this path might not be used anymore.
storage_service::notify_joined calls all subscribers `on_join_cluster`, and those, in turn do nothing:
```
bhalevy@lt scylla$ git grep -wn on_join_cluster
service/endpoint_lifecycle_subscriber.hh:37:    virtual void on_join_cluster(const gms::inet_address& endpoint) = 0;
service/qos/service_level_controller.cc:558:void service_level_controller::on_join_cluster(const gms::inet_address& endpoint) { }
service/qos/service_level_controller.hh:261:    virtual void on_join_cluster(const gms::inet_address& endpoint) override;
service/storage_proxy.cc:6563:void storage_proxy::on_join_cluster(const gms::inet_address& endpoint) {};
service/storage_proxy.hh:726:    virtual void on_join_cluster(const gms::inet_address& endpoint) override;
service/storage_service.cc:6806:                subscriber->on_join_cluster(endpoint);
service/topology_coordinator.cc:2307:    virtual void on_join_cluster(const gms::inet_address& endpoint) {}
transport/event_notifier.cc:219:void cql_server::event_notifier::on_join_cluster(const gms::inet_address& endpoint)
transport/server.hh:331:    virtual void on_join_cluster(const gms::inet_address& endpoint) override;
```

Reply:
@bhalevy it was modified as part of #15567. See also: https://github.com/scylladb/scylladb/pull/15567#discussion_r1477341663

Reply:
yes, we should eventually convert all notifications to use host_id